### PR TITLE
Fix rpc test race condition

### PR DESCRIPTION
**Pull request by kerzhner**
We are seeing timeouts in github ci such as https://github.com/statechannels/go-nitro/actions/runs/4146194983/jobs/7171543355 As is, it's hard to tell which test is causing the timeout. The hope is that using the go test timeout flag will give more debug information.

Follow up: the timeouts are caused by an rpc test race condition. The race condition has been fixed in [6a1b0e0](https://github.com/statechannels/go-nitro/pull/1125/commits/6a1b0e09b948d56a4bfccd068afe23bdadbbe4bb)

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -31,8 +31,7 @@ jobs:
         run: go build -v ./...
 
       - name: Test
-        timeout-minutes: 1
-        run: go test ./... -count=2 -shuffle=on
+        run: go test ./... -count=2 -shuffle=on -timeout 1m
```
Comment (geoknee) [inline .github/workflows/go.yml:34]: 👍

File: rpc/client.go [hunk 2]
```diff
@@ -128,9 +130,29 @@ func waitForRequest[T serde.RequestPayload, U serde.ResponsePayload](rc *RpcClie
...
-		if expectedObjectiveId == receivedObjectiveId {
+		isObjectiveExpected := false
+		for _, expectedObjectiveId := range expectedObjectiveId {
+			if receivedObjectiveId == expectedObjectiveId {
+				isObjectiveExpected = true
+			}
+		}
+		if !isObjectiveExpected {
+			err := fmt.Errorf("received unexpected objective completion notification for objective %v", receivedObjectiveId)
+			panic(err)
```
Comment (geoknee) [inline rpc/client.go:145]: Is it good to panic here?

**Review by geoknee — APPROVED** [review]
Well done tracking this down. To me, it underlines the importance of #1114 #1119.

File: rpc/client.go (hunk 2 above)
Comment (kerzhner) [reply rpc/client.go]: Good call. `WaitForObjectiveCompletion` now returns an error in [a4d5507](https://github.com/statechannels/go-nitro/pull/1125/commits/a4d550714ca6ceed03347a4dfdaac84d17c92bd0)

**Pull request merged**